bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/handler/AVMFritzBaseThingHandler.java

```diff
@@ -255,9 +256,16 @@ protected void updateLevelControl(@Nullable LevelControlModel levelControlModel)
         }
     }
 
-    private void updateDimmableLight(@Nullable LevelControlModel levelControlModel) {
-        if (levelControlModel != null) {
-            updateThingChannelState(CHANNEL_BRIGHTNESS, new PercentType(levelControlModel.getLevelPercentage()));
+    private void updateDimmableLight(@Nullable LevelControlModel levelControlModel,
+            @Nullable SimpleOnOffModel simpleOnOff) {
+        if (levelControlModel != null && simpleOnOff != null) {
+            PercentType brightness;
+            if (simpleOnOff.state) {
```

Comment:
This would prevent regressions by reverting to the old behavior when the device has a nullable `deviceModel.getSimpleOnOffUnit()`. Might not make sense if _all_ devices have this set. 
```suggestion
        if (levelControlModel != null) {
            PercentType brightness;
            if (simpleOnOff == null || simpleOnOff.state) {
```